pkg/agent/qrm-plugins/memory/dynamicpolicy/policy_advisor_handler.go

```diff
@@ -516,3 +517,165 @@ func (p *DynamicPolicy) handleAdvisorMemoryProvisions(_ *config.Configuration,
 	general.Infof("qrm: memoryProvisions: %v", memoryProvisions)
 	return nil
 }
+
+func (p *DynamicPolicy) handleNumaMemoryBalance(_ *config.Configuration,
+	_ interface{},
+	_ *dynamicconfig.DynamicAgentConfiguration,
+	emitter metrics.MetricEmitter,
+	metaServer *metaserver.MetaServer,
+	entryName, subEntryName string,
+	calculationInfo *advisorsvc.CalculationInfo, podResourceEntries state.PodResourceEntries) error {
+	advice := &types.NumaMemoryBalanceAdvice{}
+	value := calculationInfo.CalculationResult.Values[string(memoryadvisor.ControlKnobKeyBalanceNumaMemory)]
+	err := json.Unmarshal([]byte(value), advice)
+
+	if err != nil {
+		return fmt.Errorf("unmarshal %s: %s failed with error: %v",
+			memoryadvisor.ControlKnobKeyBalanceNumaMemory, value, err)
+	}
+
+	migratePagesWorkName := fmt.Sprintf("%v", memoryadvisor.ControlKnobKeyBalanceNumaMemory)
+	if p.asyncWorkers.WorkExists(migratePagesWorkName) {
+		general.Infof("work %v already exists, skip this turn", migratePagesWorkName)
+		return nil
+	}
+
+	// start an asynchronous work to migrate pages for containers
+	err = p.asyncWorkers.AddWork(migratePagesWorkName,
+		&asyncworker.Work{
+			Fn:          p.doNumaMemoryBalance,
+			Params:      []interface{}{*advice},
+			DeliveredAt: time.Now()})
+
+	if err != nil {
+		general.Errorf("add work: %s failed with error: %v", migratePagesWorkName, err)
+	}
+
+	_ = emitter.StoreInt64(util.MetricNameMemoryNumaBalance, 1, metrics.MetricTypeNameRaw)
+
+	return nil
+}
+
+type containerMigrateStat struct {
+	containerID string
+	cgroupPath  string
+	rssBefore   uint64
+	rssAfter    uint64
+}
+
+func (p *DynamicPolicy) doNumaMemoryBalance(ctx context.Context, advice types.NumaMemoryBalanceAdvice) error {
+	startTime := time.Now()
+	defer func() {
```

Comment:
there is time cost [metrics](https://github.com/kubewharf/katalyst-core/blob/a5b8bd30253a35e8842712335fde42ae75be6668/pkg/util/asyncworker/async_workers.go#L165) in the async worker framework, do we still need logic here?

Reply:
if so, we don't need this anymore. I'll remove this logic.